server/src/main/java/org/elasticsearch/search/profile/query/QueryProfileShardResult.java

```diff
@@ -35,19 +35,29 @@ public final class QueryProfileShardResult implements Writeable, ToXContentObjec
     public static final String REWRITE_TIME = "rewrite_time";
     public static final String QUERY_ARRAY = "query";
 
+    public static final String VECTOR_OPERATIONS_COUNT = "vector_operations_count";
+
     private final List<ProfileResult> queryProfileResults;
 
     private final CollectorResult profileCollector;
 
     private final long rewriteTime;
 
+    private Long vectorOperationsCount;
+
     public QueryProfileShardResult(List<ProfileResult> queryProfileResults, long rewriteTime, CollectorResult profileCollector) {
         assert (profileCollector != null);
         this.queryProfileResults = queryProfileResults;
         this.profileCollector = profileCollector;
         this.rewriteTime = rewriteTime;
     }
 
+    public QueryProfileShardResult(List<ProfileResult> queryProfileResults,
```

Comment:
You need to write in all the transport serialization and add backwards compatibility for old nodes. 

So, a new `TransportVersions` item and handling in `QueryProfileShardResult(StreamInput)` & `QueryProfileShardResult#writeTo(StreamOutput)`

Reply:
Also, need to update tests in `QueryProfileShardResultTests` to account for new items, etc.

Reply:
Updated